test/markup/cpp/template_complexity.txt

```diff
@@ -0,0 +1,20 @@
+template <class T> // comment
+auto foo(T x) { … }
+
+namespace impl {
+    template<typename T>
+    struct is_streamable<T, std::void_t<decltype(std::declval<std::wostream &>() << std::declval<T>())>> : std::true_type { };
+}
+
+// Disable overload for already valid operands.
+template<class T, class = std::enable_if_t<!impl::is_streamable_v<const T &> && std::is_convertible_v<const T &, std::wstring_view>>>
+std::wostream &operator <<(std::wostream &stream, const T &thing)
+{
+    return stream << static_cast<std::wstring_view>(thing);
+}
+
+enum struct DataHolder { }
```

Comment:
It doesn’t really matter either way but we should probably add semicolons behind this and the following type definitions to make the test cases valid C++.